packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModal.tsx

```diff
@@ -40,11 +41,17 @@ export const TransactionReviewModal = ({ decision }: TransactionReviewModalProps
     const isActionAbortable = useSelector(selectIsActionAbortable);
     const dispatch = useDispatch();
 
+    const stake = useSelector(selectStake);
+    const isSend = Boolean(send?.precomposedTx);
+    // Only one state should be available when the modal is open
+    const txInfoState = isSend ? send : stake;
+    const cancelSignTx = isSend ? cancelSignSendTx : cancelSignStakingTx;
```

Comment:
If I remember correctly, we were discussing a wrapper around this component which would decide with state to use and propage it down.

Reply:
Yes, we were. But then I saw that there is not so much logic to add, at least atm, and thought it would be an overkill.

Done. I made this component a wrapper itself in order not to change naming in other files.
https://github.com/everstake/trezor-suite/pull/4/commits/1fb21b45df06e2de53843e458784f968bc53e17e